src/lib/utils/testOpenAiAPIKey.ts

```diff
@@ -0,0 +1,23 @@
+export async function testOpenAiAPIKey(testApiKey: string): Promise<string> {
+  try {
+    const response = await fetch('https://api.openai.com/v1/chat/completions', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${testApiKey}`,
+      },
+      body: JSON.stringify({
+        model: 'gpt-4o-mini',
+        messages: [{ role: 'user', content: 'Say this is a test!' }],
+        temperature: 0.7,
+      }),
+    })
+    if (response.status === 200) {
+      return 'The API key is valid.'
+    } else {
+      return `The API key is invalid or has insufficient permissions. Status code: ${response.status}.`
+    }
+  } catch (error) {
+    return 'An error occurred while testing the API key.'
+  }
+}
```

Comment:
dont they have a test endpoint that we could use?

Reply:
I couldn't find one in their docs, so I opted for the example on their completions endpoint that they use here: https://platform.openai.com/docs/api-reference/making-requests. But let's switch it out if we find something better in the future 👍